src/Transform/ONNX/ShapeInferencePass.cpp

```diff
@@ -60,42 +51,41 @@ static SmallVector<func::FuncOp, 4> lookUpFuncsMatching(
  * purposes.
  */
 class ShapeInferencePass
-    : public PassWrapper<ShapeInferencePass, OperationPass<ModuleOp>> {
-private:
-  bool analyzeAllFunctions;
-
+    : public PassWrapper<ShapeInferencePass, OperationPass<func::FuncOp>> {
```

Comment:
it's debatable whether it's correct to make shape inference a function pass (both in ShapeInferencePass and ONNXHybridTransformPass) because of the modification of return shapes in the function signature in `inferFunctionReturnShapes()`

in principle this is unsafe if the function pass transforms functions in parallel and they look up each others' signatures in the module symbol table, but in reality shape inference doesn't look up function signatures (and likely doesn't work if there is any func::CallOp because it would probably break when the function signature changes)

Reply:
Agree. It seems that shape inference should go top-down in the call-graph. But the semantics and usage of function call in onnx model is not clear to me.

Reply:
I created issue #2185 to record this problem